cosmos/operators/gcp_cloud_run_job.py

```diff
@@ -102,7 +108,37 @@ def build_and_run_cmd(
         self.build_command(context, cmd_flags)
         self.log.info(f"Running command: {self.command}")
         result = CloudRunExecuteJobOperator.execute(self, context)
-        logger.info(result)
+
+        # Pull Google Cloud Run job logs from Google Cloud Logging to Airflow logs
+        execution_name = result["latest_created_execution"]["name"]
+        execution_time = result["latest_created_execution"]["create_time"]
+        filter_ = f'resource.type = "cloud_run_job" AND resource.labels.job_name = "{self.job_name}" AND timestamp>="{execution_time}"'
+
+        self.log.info("Attempt to retrieve logs from Google Cloud Logging")
+        time.sleep(5)  # Add sleep time to make sure all the job logs are available when we do the request
+
+        # List to store log messages
+        log_messages = []
+
+        try:
+            client = logging.Client(project=self.project_id)
+            # Search for logs associated with the job name
+            entries = client.list_entries(filter_=filter_)
+            self.log.info(f"Listing logs of the execution {execution_name}:")
+            if not entries:
+                self.log.warning("No logs found for the Cloud Run job.")
+            else:
+                for entry in entries:
+                    # Search for logs associated with the job executed
+                    if entry.labels["run.googleapis.com/execution_name"] == execution_name:
+                        log_messages.append(entry.payload)
+                        self.log.info(f"Cloud Run Log: {entry.payload}")
+                return log_messages
+
+        except GoogleCloudError as e:
+            # Catch Google Cloud-related errors (e.g., permission issues)
+            self.log.warning(f"Warning: Error retrieving logs from Google Cloud Logging: {str(e)}")
+            # Continue without raising an error, just log the warning
```

Comment:
It may be worth having a method with this logic, something like `fetch_remote_logs`